src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/MapCode/MapCodeEndpoint.cs

```diff
@@ -51,22 +54,30 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
     }
 
     public async Task<WorkspaceEdit?> HandleRequestAsync(
-        VSInternalMapCodeParams request,
+        VSInternalMapCodeParams mapperParams,
         RazorRequestContext context,
         CancellationToken cancellationToken)
     {
         // TO-DO: Apply updates to the workspace before doing mapping. This is currently
         // unimplemented by the client, so we won't bother doing anything for now until
         // we determine what kinds of updates the client will actually send us.
-        Debug.Assert(request.Updates is null);
+        Debug.Assert(mapperParams.Updates is null);
 
-        if (request.Updates is not null)
+        if (mapperParams.Updates is not null)
         {
             return null;
         }
 
+        using var ts = TelemetryScope.Create(_telemetryReporter, "mapcode", Severity.Normal,
+             new Property("mapCode.CorrelationId", mapperParams.MapCodeCorrelationId));
+
+        return await HandleMappingsAsync(mapperParams.Mappings, cancellationToken).ConfigureAwait(false);
```

Comment:
Should we pass the correlation id into this method too, and then on down into `TryMapCode` and through to Roslyn?

Reply:
Yea, I think we should have this correlation id piped through with `TrySendCSharpDelegatedMappingRequestAsync`

Reply:
Good call, done.